NwInterface.cpp

```diff
@@ -261,29 +262,25 @@ Marker NwInterface::makeVerdict(u_int8_t proto, u_int16_t vlanId,
   host = inet_ntoa(in);
   strncpy(src_host, host, sizeof(src_host)-1);
 
-  if(geoip->lookup(host, country_code, sizeof(country_code), NULL, 0)) {
-    src_maker = conf->getCountryMarker(country_code);
-    
-    strncpy(src_cc, country_code, sizeof(src_cc)-1);
+  if(geoip->lookup(host, src_ctry, sizeof(src_ctry), src_cont, sizeof(src_cont))) {
+    src_maker = conf->getMarker(src_ctry,src_cont);
     pass_local = false;
   } else {
     /* Unknown or private IP address  */
-    src_cc[0] = '\0';
+    src_ctry[0] = '\0';
     src_maker = MARKER_PASS;
   }
 
   in.s_addr = daddr;
   host = inet_ntoa(in);
   strncpy(dst_host, host, sizeof(dst_host)-1);
 
-  if(geoip->lookup(host = inet_ntoa(in), country_code, sizeof(country_code), NULL, 0)) {
-    dst_marker = conf->getCountryMarker(country_code);
-
-    strncpy(dst_cc, country_code, sizeof(dst_cc)-1);
+  if(geoip->lookup(host = inet_ntoa(in), dst_ctry, sizeof(dst_ctry), NULL, 0)) {
```

Comment:
I think that in the second look-up you should put dst_cont and sizeof(dst_cont) as parameters.
Can't we put an if before this look-up to check if already has a MARKER_DROP?
In this way we shouldn't open and search 2 times in the database

Reply:
I agree, I think we can apply this logic for ignored ports. If a port is ignored I think we can avoid the search in the database

Reply:
Right, I've added `dst_cont`, `sizeof(dst_cont)`. Adding a "short-circuit" policy would avoid doing unnecessary operations, but it would result in missing info in the stdout printing. A slight performance boost would be nice, but deciding whether to prioritize performance or logging might be a matter of discussion.